pkg/swarm/swarm.go

```diff
@@ -102,13 +110,16 @@ type Chunk interface {
 	TagID() uint32
 	WithTagID(t uint32) Chunk
 	Equal(Chunk) bool
+	Type() Type
+	WithType(t Type) Chunk
 }
 
 type chunk struct {
 	addr       Address
 	sdata      []byte
 	pinCounter uint64
 	tagID      uint32
+	chunkType  Type
```

Comment:
i would _really_ like to name this `type`, but it is a reserved word.

leaving this comment here in case anyone else has suggestions for something better than `chunk.chunkType`

Reply:
how about chunk.cType?